### PR TITLE
Add pin for Pythia 8

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by chrisburr** [comment]
@conda-forge-admin, please rerender

**Comment by chrisburr** [comment]
@conda-forge/core Ready for review/merge

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -515,6 +515,11 @@ def _gen_new_index(repodata, subdir):
             i = record['depends'].index('abseil-cpp')
             record['depends'][i] = 'abseil-cpp =20190808'
 
+        # Filter by timestamp as pythia8 also contains python bindings that shouldn't be pinned
+        if 'pythia8' in deps and record.get('timestamp', 0) < 1584264455759:
```
Comment (xhochy) [inline recipe/gen_patch_json.py]: Can you explain a bit more about this magic timestamp?

Comment (chrisburr) [reply recipe/gen_patch_json.py]: This is under the assumption that these patches apply to all conda-forge packages both present and future. I haven't seen anything confirming this but it appears to be the case?

The timestamp is just from the newest package in conda-forge at the time of me committing this line I'm confident everything currently in conda-forge won't be negatively affected by this.

In future packages might be added which depend on `pythia8` as a python library and these shouldn't inherit a pin for some random old version which had bad metadata.

Comment (xhochy) [reply recipe/gen_patch_json.py]: Sounds reasonable 👍 but I would need someone else here to approve as I cannot confirm all assumptions.

**Pull request merged**